src/main/java/com/growup/pms/task/controller/dto/response/TaskDetailResponse.java

```diff
@@ -1,16 +1,50 @@
 package com.growup.pms.task.controller.dto.response;
 
+import com.growup.pms.task.domain.Task;
+import java.time.format.DateTimeFormatter;
+import lombok.AccessLevel;
 import lombok.Builder;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
 
-@Builder
-public record TaskDetailResponse(
-        Long taskId,
-        Long statusId,
-        String userNickname,
-        String taskName,
-        String content,
-        Short sortOrder,
-        String startDate,
-        String endDate
-) {
+@Getter
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class TaskDetailResponse {
+
+    private Long taskId;
+    private Long statusId;
+    private String userNickname;
+    private String taskName;
+    private String content;
+    private Short sortOrder;
+    private String startDate;
+    private String endDate;
+
+    @Builder
+    public TaskDetailResponse(Long taskId, Long statusId, String userNickname, String taskName, String content,
+                              Short sortOrder, String startDate, String endDate) {
+        this.taskId = taskId;
+        this.statusId = statusId;
+        this.userNickname = userNickname;
+        this.taskName = taskName;
+        this.content = content;
+        this.sortOrder = sortOrder;
+        this.startDate = startDate;
+        this.endDate = endDate;
+    }
+
+    public static TaskDetailResponse of(Task task) {
+        return TaskDetailResponse.builder()
+                .taskId(task.getId())
+                .statusId(task.getStatus() != null ? task.getStatus().getId() : null)
+                .userNickname(task.getUser() != null ? task.getUser().getUsername() : null)
```

Comment:
Nit: 이 부분은 `~get`으로 도메인이 수동적이고 의미가 바로 보이지는 않아서 `isAssigned()`와 같이 명시적으로 바꿔보는 건 어떤가요? 위아래 로직에서도 그런 식으로 바꿔보면 한층 더 이해가 쉬워질 것 같습니다 :thinking: 

```suggestion
                .userNickname(task.isAssigned() ? task.getAssignee().getId() : null)
```

Reply:
statusId 도 비슷하게 수정하면 되려나요?

Reply:
괜찮은 이름이 있는지는 모르겠는데 `isStatusDefined()` 등으로 지어볼 수 있을 것 같긴 합니다 :thinking:

개인적인 의견으로는 `user`라는 키워드도 사실 일정에 와서는 작업을 수행하는 주체가 되니 `assignee`가 좀 더 적절할 것 같다는 생각이 드네요

이 부분은 영환님이 고민해보시기에 더 괜찮은 이름이나 방법이 있다면 그렇게 개선해봐도 괜찮을 것 같습니다

Reply:
네 알겠습니다! 이름은 제가 고민해보고, 수정해서 올려놓도록 하겠습니다

Reply:
:+1: 

`projectId`를 포함한 검증 로직이 반영되면 바로 머지하겠습니다 :smiley: